rest-api/instructions/models/instructions-models.js

```diff
@@ -0,0 +1,95 @@
+const db = require('../../../database/dbConfig');
+
+module.exports = {
+  editInstruction,
+  deleteInstruction,
+  addInstruction
+};
+
+async function addInstruction(id, body) {
+  return await db.transaction(async trx => {
+    try {
+      const instructions = await trx('instructions')
+        .insert(body)
+        .returning('*');
+
+      const recipe_instructions_object = instructions.map(instruction => {
+        return {
+          instruction_id: instruction.id,
+          recipe_id: id
+        };
+      });
+
+      const recipe_instructions = await trx('recipe_instructions').insert(
+        recipe_instructions_object
+      );
+
+      if (recipe_instructions.rowCount) {
+        return { message: 'Instruction added sucessfully.' };
+      } else {
+        throw { error: 'Instruction not added' };
+      }
+    } catch (error) {
+      console.log(error);
+      throw error;
+    }
+  });
+}
+
+async function getSteps(id) {
+  const instructions = await db('recipe_instructions')
+    .join(
+      'instructions',
+      'instructions.id',
+      'recipe_instructions.instruction_id'
+    )
+    .select('instructions.text', 'instructions.id')
+    .where('recipe_instructions.recipe_id', id)
+    .orderBy('recipe_instructions.instruction_id')
+    .map((ing, i) => {
+      return { ...ing, step: i + 1 };
+    });
+
+  return instructions;
+}
+
+async function deleteInstruction(id, body) {
+  const instructions = await getSteps(id);
+
+  const filtered = instructions.find(i => {
+    if (i.step === body.step) {
+      return i.id;
+    }
+  });
+
```

Comment:
Hmm... a question about this one. What if more than one recipe points to the same instruction?

Reply:
Or there are duplicate instructions?

Reply:
Yes, to save time I decided it would be better since people can just edit instead of creating and deleting

Reply:
That way we save resources, since it would take 2 requests instead of 1 (this way)

Reply:
Gotcha. :))